charts/osm/templates/mutatingwebhook.yaml

```diff
@@ -10,7 +10,7 @@ webhooks:
     service:
       name: osm-controller
       namespace: {{.Release.Namespace}}
-      path: /mutate
+      path: /create/pod
```

Comment:
why is this being changed?
The sidecar injector webhook is only ever going to mutate pod creation requests, so just `/mutate` on the webhook suffices to convey the mutation is on the pod creation. L36-39 indicate that the mutation will be performed for pod creation requests only.

Reply:
The handler for this endpoint is explicitly only for pod creation events -- we are on the same page on that.

The naming of the endpoint however is very generic - `/mutate` does not say much about what is going to happen.
But since the handler will only work for pod creation - `/create/pod` adds sufficient context as to what this is going to do.  I believe that with this change, should we decide to broaden the scope of the MutatingWebhookConfig to other events, the specificity of this endpoint would give us a hint that the actual handler would nede to be changed as well.

Reply:
@shashankram I renamed this endpoint to `/mutate-pod-creation`, which, from our conversation, we both agree that it contextualizes the limited functionality the underlying handler has.
Thank you for helping me make this better!